services/cd-service/pkg/service/overview.go

```diff
@@ -312,6 +312,55 @@ func (o *OverviewServiceServer) GetOverview(
 	return o.getOverviewDB(ctx, o.Repository.State())
 }
 
+func (o *OverviewServiceServer) GetAllAppLocks(ctx context.Context,
+	in *api.GetAllAppLocksRequest) (*api.GetAllAppLocksResponse, error) {
+
+	span, ctx := tracer.StartSpanFromContext(ctx, "GetAllAppLocks")
+	defer span.Finish()
+
+	return db.WithTransactionT(o.DBHandler, ctx, 2, true, func(ctx context.Context, transaction *sql.Tx) (*api.GetAllAppLocksResponse, error) {
+		state := o.Repository.State()
+
+		allAppNames, err := state.GetApplications(ctx, transaction)
+		if err != nil {
+			return nil, err
+		}
+
+		response := api.GetAllAppLocksResponse{
+			AllAppLocks: make(map[string]*api.AllAppLocks),
+		}
+		for _, appName := range allAppNames {
+			if err != nil {
+				return nil, err
+			}
+			appLocks, err := o.DBHandler.DBSelectAllActiveAppLocksForApp(ctx, transaction, appName)
```

Comment:
Can't we get all app locks for all apps at once instead of iterating over apps? It might cause performance issues.